go/pkg/cas/upload_test.go

```diff
@@ -105,35 +107,49 @@ func TestFS(t *testing.T) {
 	}
 }
 
-func TestChecks(t *testing.T) {
+func TestSmallBlobs(t *testing.T) {
 	t.Parallel()
 	ctx := context.Background()
 
 	var mu sync.Mutex
 	var gotDigestChecks []*repb.Digest
-	var gotRequestSizes []int
-	var gotScheduleUploadCalls []*uploadItem
+	var gotDigestCheckRequestSizes []int
+	var gotUploadBlobReqs []*repb.BatchUpdateBlobsRequest_Request
+	failCBlob := true // blob "c" is uploaded below.
```

Comment:
what about `failBlob := map[dg]error`?

Reply:
could you elaborate? what do we gain from the additional complexity?

Reply:
flexibility?
Tbh I'm thinking of the case where fakeCAS is reused. I guess as long as it's test specific, that doesn't matter.